product-service/src/utils/product-validate.ts

```diff
@@ -0,0 +1,33 @@
+import { MESSAGES, STATUS_CODES, COMMON_PATH } from '../constants/constants';
+import { IProduct, SKATE_TYPES } from '../models/product.model';
+import { MyError } from './error';
+
+export const productValidate = (product): IProduct => {
+  const { description, count, title, price, image } = product;
+
+  if (!count || typeof +count !== "number") {
+    throw new MyError(STATUS_CODES.PRODUCT_DATA_IS_INVALID, `${MESSAGES.PRODUCT_DATA_IS_INVALID}: count`);
+  }
+
+  if (!title || typeof title !== "string") {
+    throw new MyError(STATUS_CODES.PRODUCT_DATA_IS_INVALID, `${MESSAGES.PRODUCT_DATA_IS_INVALID}: title`);
+  }
+
+  if (!price || typeof +price !== "number") {
+    throw new MyError(STATUS_CODES.PRODUCT_DATA_IS_INVALID, `${MESSAGES.PRODUCT_DATA_IS_INVALID}: price`);
+  }
+
+  const validatingProduct = { ...product };
+
+  if ( !description ||
+    description !== SKATE_TYPES.DERBY ||
+    description !== SKATE_TYPES.RETRO ) {
+    validatingProduct.description = SKATE_TYPES.DERBY
+  }
```

Comment:
It's not intuitive. I even thought that it's bug. There is "string" type for description field in your swagger docs. Please, fix it.

P.S. It's not affect the final score, but still...

Reply:
Why do you restrict the product description to certain values? It can be any text, so the type should be just `string`

Reply:
When I try to create a product, the description is just being overridden without any notifications. It will affect the final score because POST method doesn't work as expected

Reply:
I agree with you. I'll fix it for the next task. Thank for feedback